core/src/test/scala/org/apache/spark/sql/delta/DeltaRestoreTableSuite.scala

```diff
@@ -0,0 +1,148 @@
+    .withSequenceColumn("seq")
+    .withColumnRenamed("id", "new_id")
+  private lazy val allDatasets = Seq(versionZeroData, versionOneData)
+
+  test("basic case - Scala restore table using version - Non-partitioned") {
+    withMultiVersionedDeltaTable(allDatasets) { deltaPath =>
+
+      DeltaTableForPath(deltaPath).restore(0)
+
+      testRestoredTable(deltaPath)
+    }
+  }
+
+  test("basic case - Scala restore table using version - Partitioned") {
+    withMultiVersionedDeltaTable(allDatasets, Some("seq")) { deltaPath =>
+
+      DeltaTableForPath(deltaPath).restore(0)
+
+      testRestoredTable(deltaPath)
+    }
+  }
+
+  test("basic case - Scala restore table using timestamp - Non-partitioned") {
+    withMultiVersionedDeltaTable(allDatasets) { deltaPath =>
+      val zeroVersionTs = new Timestamp(
+        DeltaLog.forTable(spark, deltaPath).getSnapshotAt(0).timestamp)
+
+      DeltaTableForPath(deltaPath).restore(zeroVersionTs)
+
+      testRestoredTable(deltaPath)
+    }
+  }
+
+  test("basic case - Scala restore table using timestamp - Partitioned") {
+    withMultiVersionedDeltaTable(allDatasets, Some("seq")) { deltaPath =>
+      val zeroVersionTs = new Timestamp(
+        DeltaLog.forTable(spark, deltaPath).getSnapshotAt(0).timestamp)
+
+      DeltaTableForPath(deltaPath).restore(zeroVersionTs)
+
+      testRestoredTable(deltaPath)
+    }
+  }
+
+  test("basic case - Scala restore failed if any data file is missed" +
+      " and spark.sql.files.ignoreMissingFiles=false") {
+    withMultiVersionedDeltaTable(allDatasets) { deltaPath =>
+      import testImplicits._
+      // Remove one data files from table
+      val file = DeltaLog.forTable(spark, deltaPath).getSnapshotAt(0).allFiles.map(_.path).head()
+      FileUtils.deleteQuietly(new File(new Path(deltaPath, file).toUri.toString))
+
+      val ex = intercept[IllegalArgumentException](DeltaTableForPath(deltaPath).restore(0))
+      assert(ex.getMessage.contains(file), s"Error message doesn't contain missed file $file."
+        + s"Error message:\n ${ex.getMessage}")
+    }
+  }
+
+  test("basic case - Scala restore failed if any data file is missed" +
+      " and spark.sql.files.ignoreMissingFiles=true") {
+    withSQLConf(SQLConf.IGNORE_MISSING_FILES.key -> "true") {
+      withMultiVersionedDeltaTable(allDatasets) { deltaPath =>
+        import testImplicits._
+        // Remove one data files from table
+        val file = DeltaLog.forTable(spark, deltaPath).getSnapshotAt(0).allFiles.map(_.path).head()
+        FileUtils.deleteQuietly(new File(new Path(deltaPath, file).toUri.toString))
+
+        DeltaTableForPath(deltaPath).restore(0)
+
+        testRestoredTable(deltaPath, validateData = false)
+      }
+    }
+  }
+
+  test("basic case - Scala restore failed to restore incorrect version") {
+    withMultiVersionedDeltaTable(allDatasets) { deltaPath =>
+
+      intercept[IllegalArgumentException](DeltaTableForPath(deltaPath).restore(Long.MaxValue))
```

Comment:
any checks on the error message? it should mention the right set of things in the error message.